### PR TITLE
[FIX] stock_move_packaging_qty: add packaging qty reserved

### PR DESCRIPTION
**Pull request by EmilioPascual**
- Add packaging qty reserved in stock move line with compute to know packaging reserved when reserved.

@yajo @Shide @rafaelbn @Gelojr please review.

MT-5861 @moduon

### CODE REVIEW

File: stock_move_packaging_qty/models/stock_move_line.py [hunk 1]
```diff
@@ -9,11 +10,52 @@ class StockMoveLine(models.Model):
...
+                line.product_packaging_qty_reserved = False
+                continue
+            packaging_uom = line.product_packaging_id.product_uom_id
+            packaging_uom_qty = line.product_uom_id._compute_quantity(
+                line.reserved_uom_qty, packaging_uom
+            )
+            line.product_packaging_qty_reserved = float_round(
+                packaging_uom_qty / line.product_packaging_id.qty,
+                precision_rounding=packaging_uom.rounding,
+            )
```
Comment (Shide) [inline stock_move_packaging_qty/models/stock_move_line.py]: UoM of product_packaging is always the same as product UoM

```suggestion
            line.product_packaging_qty_reserved = float_round(
                line.reserved_uom_qty / line.product_packaging_id.qty,
                precision_rounding=line.product_uom_id.rounding,
            )
```

File: stock_move_packaging_qty/models/stock_move_line.py [hunk 2]
```diff
@@ -9,11 +10,52 @@ class StockMoveLine(models.Model):
...
+                line.product_packaging_qty_done = False
+                continue
+            packaging_uom = line.product_packaging_id.product_uom_id
+            packaging_uom_qty = line.product_uom_id._compute_quantity(
+                line.qty_done, packaging_uom
+            )
+            line.product_packaging_qty_done = float_round(
+                packaging_uom_qty / line.product_packaging_id.qty,
+                precision_rounding=packaging_uom.rounding,
+            )
```
Comment (Shide) [inline stock_move_packaging_qty/models/stock_move_line.py]: I think product_packaging_qty_done should not be computed

**Review by Shide — CHANGES_REQUESTED** [review]

File: stock_move_packaging_qty/views/stock_move_line_view.xml [hunk 3]
```diff
@@ -23,6 +23,23 @@
...
+            <xpath expr="//label[@for='reserved_uom_qty']" position="before">
+                <label
+                    for="product_packaging_qty_reserved"
+                    attrs="{'invisible': [('state', '=', 'done')]}"
+                />
+                <div class="o_row" attrs="{'invisible': [('state', '=', 'done')]}">
+                    <field
+                        name="product_packaging_qty_reserved"
+                        attrs="{'invisible': [('product_packaging_id', '=', False)]}"
+                        groups="product.group_stock_packaging"
```
Comment (yajo) [inline stock_move_packaging_qty/views/stock_move_line_view.xml]: issue: add this `groups` to the parent div.

File: stock_move_packaging_qty/views/stock_move_line_view.xml [hunk 4]
```diff
@@ -23,6 +23,23 @@
         <field name="model">stock.move.line</field>
         <field name="inherit_id" ref="stock.view_move_line_mobile_form" />
         <field name="arch" type="xml">
+            <xpath expr="//label[@for='reserved_uom_qty']" position="before">
+                <label
+                    for="product_packaging_qty_reserved"
+                    attrs="{'invisible': [('state', '=', 'done')]}"
```
Comment (yajo) [inline stock_move_packaging_qty/views/stock_move_line_view.xml:29]: issue: missing `groups`

File: stock_move_packaging_qty/models/stock_move_line.py (hunk 1 above)
Comment (yajo) [reply stock_move_packaging_qty/models/stock_move_line.py]: However, it doesn't have to be the same as the UoM in the `stock.move.line`, right? That is what he's handling here AFAICS.

File: stock_move_packaging_qty/models/stock_move_line.py [hunk 5]
```diff
@@ -9,11 +10,52 @@ class StockMoveLine(models.Model):
...
+        compute="_compute_product_packaging_qty_done",
+        store=True,
+        readonly=False,
     )
 
+    @api.depends("product_packaging_id", "product_uom_id", "reserved_uom_qty")
+    def _compute_product_packaging_qty_reserved(self):
+        """Get the quantity done in product packaging."""
+        for line in self:
+            if not line.product_packaging_id or not line.product_packaging_id.qty:
```
Comment (yajo) [inline stock_move_packaging_qty/models/stock_move_line.py]:
```suggestion
            if not line.product_packaging_id.qty:
```

**Review by Shide — APPROVED** [review]

File: stock_move_packaging_qty/models/stock_move_line.py [hunk 6]
```diff
@@ -9,11 +10,29 @@ class StockMoveLine(models.Model):
...
+                line.product_packaging_qty_reserved = False
+                continue
+            line.product_packaging_qty_reserved = float_round(
+                line.reserved_uom_qty / line.product_packaging_id.qty,
+                precision_rounding=line.product_uom_id.rounding,
+            )
```
Comment (yajo) [inline stock_move_packaging_qty/models/stock_move_line.py]: issue: IIUC you have to keep in mind the conversion between the line uom and the packaging uom.

For that, you can get the data from `reserved_qty` instead, [which is always in the same UoM as the product](https://github.com/odoo/odoo/blob/1d55662f424911873cb2ddcb99e3caf37f43294e/addons/stock/models/stock_move_line.py#L128-L129) (and thus the packaging).

```suggestion
            line.product_packaging_qty_reserved = float_round(
                line.reserved_qty / line.product_packaging_id.qty,
                precision_rounding=line.product_uom_id.rounding,
            )
```

File: stock_move_packaging_qty/models/stock_move_line.py [hunk 7]
```diff
@@ -9,11 +10,29 @@ class StockMoveLine(models.Model):
...
+        help="Product packaging quantity reserved.",
+        compute="_compute_product_packaging_qty_reserved",
+        store=True,
+    )
     product_packaging_qty_done = fields.Float(
         string="Done Pkg. Qty.",
         help="Product packaging quantity done.",
     )
 
+    @api.depends("product_packaging_id", "product_uom_id", "reserved_uom_qty")
```
Comment (yajo) [inline stock_move_packaging_qty/models/stock_move_line.py]: issue: see below

```suggestion
    @api.depends("product_packaging_id", "product_uom_id", "reserved_uom_qty", "reserved_qty")
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by Shide** [comment]
@yajo all fixes attended

**Review by yajo — APPROVED** [review]
/ocabot merge minor

**Pull request merged**